### PR TITLE
feature/add-MakeClassAbstractSniff

### PR DESCRIPTION
**Pull request by alexndlm**
This sniff require class to be abstract. It is will be useful for TestCase classes.

### CODE REVIEW

File: src/Sniffs/Classes/MakeClassAbstractSniff.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+declare(strict_types=1);
+
+namespace EonX\EasyQuality\Sniffs\Classes;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use SlevomatCodingStandard\Helpers\NamespaceHelper;
+
+class MakeClassAbstractSniff implements Sniff
```
Comment (roman-eonx) [inline src/Sniffs/Classes/MakeClassAbstractSniff.php:11]: Please add this new sniff to the docs with an example of applying it to TestCase classes.

**Review by roman-eonx — CHANGES_REQUESTED** [review]

File: src/Sniffs/Classes/MakeClassAbstractSniff.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int $clasPointer The position of the current token in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $clasPointer): void
```
Comment (ERuban) [inline src/Sniffs/Classes/MakeClassAbstractSniff.php:31]: `$classPointer`

File: src/Sniffs/Classes/MakeClassAbstractSniff.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    /**
+     * @var mixed[]
+     */
+    public $applyTo = [];
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
```
Comment (ERuban) [inline src/Sniffs/Classes/MakeClassAbstractSniff.php:26]: Not needed

File: src/Sniffs/Classes/MakeClassAbstractSniff.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+     * @param int $clasPointer The position of the current token in the stack passed in $tokens.
+     *
+     * @return void
```
Comment (ERuban) [inline src/Sniffs/Classes/MakeClassAbstractSniff.php:29]: Not needed

File: src/Sniffs/Classes/MakeClassAbstractSniff.php [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                $phpcsFile->fixer->replaceToken($finalTokenPtr, 'abstract');
+            } else {
+                $phpcsFile->fixer->addContentBefore($clasPointer, 'abstract ');
+            }
+            $phpcsFile->fixer->endChangeset();
+        }
+    }
+
+    /**
+     * @param string $classFqn
```
Comment (ERuban) [inline src/Sniffs/Classes/MakeClassAbstractSniff.php:60]: Not needed

**Review by ERuban — CHANGES_REQUESTED** [review]

File: src/Sniffs/Classes/MakeClassAbstractSniff.php (hunk 2 above)
Comment (ERuban) [reply src/Sniffs/Classes/MakeClassAbstractSniff.php:31]: Also, usually this parameter named as `$stackPtr`

File: src/Sniffs/Classes/MakeClassAbstractSniff.php (hunk 3 above)
Comment (ERuban) [reply src/Sniffs/Classes/MakeClassAbstractSniff.php:26]: I think we can get rid of these phpdocs as we already have it in the parent class

**Review by ERuban — CHANGES_REQUESTED** [review]

File: src/Sniffs/Classes/MakeClassAbstractSniff.php (hunk 1 above)
Comment (alexndlm) [reply src/Sniffs/Classes/MakeClassAbstractSniff.php:11]: Done.

File: src/Sniffs/Classes/MakeClassAbstractSniff.php (hunk 2 above)
Comment (alexndlm) [reply src/Sniffs/Classes/MakeClassAbstractSniff.php]: Done.

File: src/Sniffs/Classes/MakeClassAbstractSniff.php (hunk 3 above)
Comment (alexndlm) [reply src/Sniffs/Classes/MakeClassAbstractSniff.php]: Done.

File: src/Sniffs/Classes/MakeClassAbstractSniff.php (hunk 4 above)
Comment (alexndlm) [reply src/Sniffs/Classes/MakeClassAbstractSniff.php]: Done.

File: src/Sniffs/Classes/MakeClassAbstractSniff.php (hunk 5 above)
Comment (alexndlm) [reply src/Sniffs/Classes/MakeClassAbstractSniff.php]: Done.

**Review by roman-eonx — APPROVED** [review]

**Review by ERuban — APPROVED** [review]

**Pull request merged**